### DISCUSSION TITLE
float type modelTpl fails when calling pinocchio::urdf::buildModel

### DISCUSSION THREAD

**Discussion by modu-robotics** [stack-of-tasks/pinocchio#2038] [Q&A] [answered]
## Bug description

When I call `buildModel<float>` for `ModelTpl<float>` with a `urdf` file, the code has a compilation error saying:
```
error: no matching function for call to 'parseRootTree'
      details::parseRootTree(filename, visitor);
candidate function not viable: no known conversion from 'details::UrdfVisitor<float, 0, pinocchio::JointCollectionDefaultTpl>' to 'pinocchio::urdf::details::UrdfVisitorBase &' (aka 'UrdfVisitorBaseTpl<double, 0> &') for 2nd argument
      void PINOCCHIO_DLLAPI parseRootTree(const std::string & filename,
```

I think this is a bug since `parseRootTree` takes `UrdfVisitorBase` instead of `UrdfVisitorBaseTpl`.

## Expected behavior

Create float type model and load urdf file.

## Reproduction steps

```
 pinocchio::ModelTpl<float> model_f;
 pinocchio::urdf::buildModel<float>(urdf_file, model_f);
```

### System

- OS: [e.g. Mac]
- Pinocchio version: Installed it from brew

**Comment by jcarpent** [comment, marked as answer]
URDF parsers (urdfdom) only consider double types.
I would advise you to load using the ModelTpl<double> type, and then cast it to float type using:
```cpp
ModelTpl<double> model_double;
// load the model stored in model_double
ModelTpl<float> model_float = model_double.cast<float>();
```
